### PR TITLE
Migrate to GINKO/v2

### LINKED ISSUE #94: Migrate GINKO framework to V2
Spin off https://github.com/kubernetes-sigs/kueue/pull/89#discussion_r819686679

We should consider upgrading ginko to v2 as stated in their logs 

-  Learn more at: https://github.com/onsi/ginkgo/blob/ver2/docs/MIGRATING_TO_V2.md#removed-custom-reporters

for now we can work around by using `ACK_GINKGO_DEPRECATIONS=1.16.5` env var when running the integration tests

### PR DESCRIPTION
**Pull request by ArangoGutierrez**
Signed-off-by: Carlos Eduardo Arango Gutierrez <carangog@redhat.com>

#### Which issue(s) this PR fixes:

Fixes #94

### CODE REVIEW

**Comment by ArangoGutierrez** [comment]
as part of investigating #101

**Comment by ArangoGutierrez** [comment]
> /retest-required

Prow flake, run never started and hit the time limit

File: test/integration/scheduler/scheduler_test.go [hunk 1]
```diff
@@ -17,7 +17,7 @@ limitations under the License.
 package scheduler
 
 import (
-	"github.com/onsi/ginkgo"
+	ginkgo "github.com/onsi/ginkgo/v2"
```
Comment (alculquicondor) [inline test/integration/scheduler/scheduler_test.go:20]: I suppose this is not needed.

Comment (ArangoGutierrez) [reply test/integration/scheduler/scheduler_test.go:20]: ups, was needed at the beginning of the reorg , removing....

Comment (ArangoGutierrez) [reply test/integration/scheduler/scheduler_test.go]: done

**Comment by ArangoGutierrez** [comment]
> /hold

Requested change is done ✅ 

/hold cancel

**Pull request merged**